ssl/ssl_ciph.c

```diff
@@ -1979,8 +2036,9 @@ int ssl_cipher_get_cert_index(const SSL_CIPHER *c)
         return SSL_PKEY_DSA_SIGN;
     else if (alg_a & SSL_aRSA)
         return SSL_PKEY_RSA_ENC;
-    else if (alg_a & SSL_aGOST01)
-        return SSL_PKEY_GOST01;
+    else if (alg_a & (SSL_aGOST01 | SSL_aGOST12))
+        return SSL_PKEY_GOST_EC;
+
```

Comment:
Is this right? As I understood things, for GOST12 you can use GOST12_512, GOST12_256 or GOST01 certs...*but* for GOST01 surely you can *only* use a GOST01 cert? So wouldn't that be:
````
else if (alg_a & SSL_aGOST12)
    return SSL_PKEY_GOST_EC;
else if (alg_a & SSL_aGOST01)
    return SSL_PKEY_GOST_01;
````